src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveResourceService.cs

```diff
@@ -113,10 +114,26 @@ public async Task<RetrieveResourceResponse> GetInstanceResourceAsync(RetrieveRes
             }
         }
 
-        private static Stream ResetDicomFileStream(Stream stream)
+        private static string GetResponseTransferSyntax(string requestedTransferSyntax, VersionedInstanceIdentifier identifier)
+        {
+            bool isOriginalTransferSyntaxRequested = DicomTransferSyntaxUids.IsOriginalTransferSyntaxRequested(requestedTransferSyntax);
+            if (isOriginalTransferSyntaxRequested)
+            {
+                return GetOriginalTransferSyntaxWithBackCompat(requestedTransferSyntax, identifier);
+            }
+            return requestedTransferSyntax;
+        }
+
+        /// <summary>
+        /// Existing dicom files(as of Feb 2022) do not have transferSyntax stored. 
+        /// Untill we backfill those files, we need this existing buggy fall back code: requestedTransferSyntax can be "*" which is the wrong content-type to return
+        /// </summary>
+        /// <param name="requestedTransferSyntax"></param>
+        /// <param name="identifier"></param>
+        /// <returns></returns>
+        private static string GetOriginalTransferSyntaxWithBackCompat(string requestedTransferSyntax, VersionedInstanceIdentifier identifier)
         {
-            stream.Seek(offset: 0, SeekOrigin.Begin);
-            return stream;
+            return string.IsNullOrEmpty(identifier.Properties?.TransferSyntaxUid) ? requestedTransferSyntax : identifier.Properties.TransferSyntaxUid;
```

Comment:
identifier.Properties can be null should we do like
`identifier.Properties != null && !string.IsNullOrEmpty(identifier.Properties.TransferSyntaxUid) ? identifier.Properties.TransferSyntaxUid : requestedTransferSyntax`

Reply:
Same thing. Yours is inverse of what I have. Any reason to convert it like yours?

Reply:
I like that the original is more succinct 😄

Reply:
No worries, yours is fine